File: lecture_007(Strings)/l_009_compressastring.cpp
#include<iostream>
#include<string>
using namespace std;

void compress1(string s) {
    int count = 1;
    for(int i=0; i<s.size()-1; i++) {
        char r1 = s[i];
        char r2 = s[i+1];
        if(r1 != r2) {
            cout<<r1;
            if(count>1) {
                cout<<count; 
            }
            count=1;
        }
        else {
            count++;
        }
    }
    cout<<s[s.size()-1];
    if(count>1) {
        cout<<count;
    }
}
string compress2(string s) {
	string sb;	
    int count = 1;
    for(int i=0; i<s.length()-1; i++) {
        char r1 = s[i];
        char r2 = s[i+1];
        if(r1 != r2) {
            sb = sb + r1;
            sb = sb + to_string(count);
            count=0;
        }
        count++;
    }
    sb += s[(s.length()-1)];
    sb += to_string(count);
    return sb;
}
int main() {
    string s;
    cin>>s;
    //cout<<compress1(s);
    cout<<compress2(s);

    return 0;
}

File: lecture_007(Strings)/l_009_compressastring.java
import java.util.Scanner;
public class l_009_compressastring {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scn.nextLine();
        //compress1(s);
        compress2(s);
    }
    public static void compress1(String s) {
        int count = 1;
        for(int i=0; i<s.length()-1; i++) {
            char r1 = s.charAt(i);
            char r2 = s.charAt(i+1);
            if(r1 != r2) {
                System.out.print(r1);
                if(count>1) {
                    System.out.print(count); 
                }
                count=1;
            }
            else {
                count++;
            }
        }
        System.out.print(s.charAt(s.length()-1));
        if(count>1) {
            System.out.print(count);
        }
    }
    public static String compress2(String s) {
		StringBuilder sb = new StringBuilder();
		int count = 1;
		for(int i=0; i<s.length()-1; i++) {
			char r1 = s.charAt(i);
			char r2 = s.charAt(i+1);
			if(r1 != r2) {
				sb.append(r1);
				sb.append(count);
				count=0;
			}
			count++;
		}
		sb.append(s.charAt(s.length()-1));
		sb.append(count);
		return sb.toString();
	}
}